secant/UI Components/Chips/EnumeratedChip.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  EnumeratedChip.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 10/19/21.
+//
+
+import SwiftUI
+
+struct EnumeratedChip: View {
+    @Clamping(1...24)
+    var index: UInt = 1
+
+    var text: String
+
+    var body: some View {
+        NumberedText(
+            number: index,
+            text: text
+        )
+        .frame(
+            minWidth: 0,
+            maxWidth: .infinity,
+            minHeight: 30,
+            maxHeight: .infinity,
+            alignment: .leading
+        )
+        .padding(.horizontal, 16)
+        .padding(.vertical, 8)
+
+        .background(
+            Asset.Colors.Buttons.primaryButtonPressed.color
+        )
+        .cornerRadius(6)
+    }
+}
+
+struct NumberedText: View {
+    @Clamping(1...24) var number: UInt = 1
+
+    var text: String
+
+    @ViewBuilder var numberedText: some View {
+        Text(number.superscriptRepresentation)
+            .foregroundColor(
+                Asset.Colors.Text.highlightedSuperscriptText.color
+            )
+            .font(
+                .custom(
+                    FontFamily.Roboto.black.name,
+                    size: 24
+                )
+            ) +
+        Text(" \(text)")
+            .foregroundColor(
+                Asset.Colors.Text.button.color
+            )
+            .font(
+                .custom(
+                    FontFamily.Rubik.regular.name,
+                    size: 15
+                )
+            )
+    }
+
+    var body: some View {
+        numberedText
+            .layoutPriority(1)
+            .fixedSize(horizontal: false, vertical: false)
+    }
+}
+
+struct EnumeratedChip_Previews: PreviewProvider {
+    private static var threeColumnGrid = Array.init(
+        repeating: GridItem(
+            .flexible(minimum: 60, maximum: 120),
+            spacing: 15,
+            alignment: .topLeading
+            ),
+        count: 3
+        )
+
+    static var words = [
+        "pyramid", "negative", "page",
+        "crown", "secret", "zebra"
+    ]
+    @ViewBuilder static var grid: some View {
+        LazyVGrid(
+            columns: threeColumnGrid,
+            alignment: .leading,
+            spacing: 15
+        ) {
+            ForEach(Array(zip(words.indices, words)), id: \.1) { i, word in
+                EnumeratedChip(
+                    index: UInt((i + 1)),
```

Comment:
This is where you need to reinitialize a new Value that is extra work.    Above in the clamped value of the View, if it was an Int, you wouldn't need to do this.

This view is another good example that would also read better and make the view easier to understand if it was inlined.  

``` swift
EnumeratedChip(index: i + 1, text: word)
    .frame(minWidth: 0, maxWidth: .infinity, minHeight: 50)
```